TempoChanges.qml

```diff
@@ -172,6 +172,50 @@ MuseScore {
             }
       }
 
+      function deltaTempo(fraction)
+      {
+        // fraction is the current fraction of the number of ticks in the range 0.0 - 1.0
+        //
+        // The early/late linearity slider also ranges from 0.0 to 1.0, or just shy of that to avoid exceptions.
+        // With the slider at 0 we would like all of the tempo change to be applied immediately.
+        // With the slider at at 1/4 we would like the mid tempo to be reached 1/4 of the way through the change.
+        // With the slider at at 1/2 we would like the mid tempo to be reached 1/2 way through the change, etc.
+        //
+        // We can do this by raising the fraction to a certain power.
+        ///
+        // by inspection (fraction is f):
+        // f ^ 3 maps 7/8 => 1/2
+        // f ^ 2 maps 3/4 => 1/2
+        // f ^ 1 maps 1/2 => 1/2 etc.
+        //
+        // so we want a function that maps linearity l to power p:
+        // l   => p
+        // 7/8 => 3
+        // 3/4 => 2
+        // 1/2 => 1
+        // assumption: this holds for l = 1/4, 1/8 etc.
+        //
+        // by inspection the inverse function p => l is: l = 1 - 1/(2^p)
+        // so we need to calculate p given l:
+        //
+        //     l = 1 - 1/(2^p)
+        //
+        //     l + 1/(2^p) = 1
+        //
+        //     1/(2^p) = 1 - l
+        //
+        //     1 = (2^p)(1 - l)
+        //
+        //     1/(1 - l) = 2^p
+        //
+        //     log(1/(1 - l)) = log(2^p) = p*log(2)
+        //
+        //     log(1/(1 - l))/log(2) = p
+        //
+        var power = Math.log(1 / (1 - linearity.value)) / Math.log(2);
```

Comment:
Shouldn't it be Math.log(1 / linearity.value) / Math.log(2); to not work reverse compared to how it does now?

Reply:
I'm pretty sure it's correct, and I've manually tested both early and late for both rit. and accel. and saw reasonable-looking values in all cases, of course you should not take my word for that.

Re the 75% limit, could easily do, but then there will always be someone who wants a little more, maybe 95%?

Reply:
Here's a link to a graph of the function for linearity values 1/16, 1/8, 1/4, 1/2, 3/4, 7/8 and 15/16:
https://www.desmos.com/calculator/wq8btekqba
you can see that for i.e. 1/4 (slider at 1/4) the curve cuts the y = 1/2 line at 0.25.

Reply:
hold on this, there seems to be a mistake in my maths, the general behaviour is correct, but the precise values are wrong, i.e. slider at 1/4 doesn't map an input fraction of 1/4 to 1/2. I'll try to fix now.